lib/apiservers/portlayer/restapi/handlers/task_handlers.go

```diff
@@ -81,6 +83,20 @@ func (handler *TaskHandlersImpl) JoinHandler(params tasks.JoinParams) middleware
 		StopSignal: params.Config.StopSignal,
 	}
 
+	// parsing user
+	if params.Config.User != "" {
+		parts := strings.Split(params.Config.User, ":")
+		var user, group string
+		if len(parts) > 0 {
+			user = parts[0]
+		}
+		if len(parts) > 1 {
+			group = parts[1]
+		}
+		sessionConfig.User = user
```

Comment:
Are sessionConfig User and Group are empty before this moment? If so, then you don't need those temporary variables.